sql/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -130,9 +134,24 @@ private void expandColumn(AnalyzedColumnDefinition column) {
         }
     }
 
-    public List<String> primaryKeys() {
+    public Set<String> notNullColumns() {
+        if (notNullColumns == null) {
+            notNullColumns = new LinkedHashSet<>();
+            for (AnalyzedColumnDefinition column : columns) {
+                String fqn = column.ident().fqn();
+                if (column.isNotNull() && !primaryKeys().contains(fqn)) {
+                    if (!notNullColumns.contains(fqn)) {
```

Comment:
If the order is important then please add a comment here explaining this, otherwise it's not obvious